lib/index.js

```diff
@@ -327,7 +327,13 @@ function serializeStruct(schema, obj, writer) {
         obj.borshSerialize(writer);
         return;
     }
-    const structSchema = schema.get(obj.constructor);
+    let structSchema = undefined;
+    for (const key of schema.keys()) {
+        if (key.name == obj.constructor.name) {
+            structSchema = schema.get(key);
+            break;
+        }
+    }
```

Comment:
A bit more idiomatic JS without the local mutated variable:

```suggestion
const structSchema = schema.get(schema.keys().find((classConstructor) => (classConstructor.name === obj.constructor.name)));
```

Array.find() returns `undefined` if no elements of the array match the predicate